packages/services/f-analytics/src/store/analytics.module.js

```diff
@@ -35,30 +43,64 @@ export default {
             requestId: '',
             orientation: '',
             display: ''
-        }
+        },
+        events: []
     }),
 
     actions: {
-        updatePlatformData: ({ commit }, platformData) => {
-            commit('pushPlatformData', platformData);
+        [PUSH_PLATFORM_DATA]: ({ commit, state }, platformData) => {
+            if (platformData) {
+                commit(PUSH_PLATFORM_DATA, platformData);
+            }
+
+            if (isDataLayerPresent()) {
+                window.dataLayer.push({ platformData: { ...state.platformData } });
+            }
+        },
+
+        [PUSH_USER_DATA]: userData => {
```

Comment:
As we don't manipulate the state here we shouldn't have this action. This logic should live in the methods